src/data_conn.c

```diff
@@ -1706,10 +1719,38 @@ uzfs_zvol_rebuild_scanner(void *arg)
 						    zinfo->name);
 						goto exit;
 					}
+					mutex_enter(&zv->rebuild_mtx);
+					if (zinfo->is_snap_inprogress) {
+						sleep(1);
+						LOG_INFO("waiting for snapshot"
+						    "to finish");
+						mutex_exit(&zv->rebuild_mtx);
+						goto retry;
+					}
+					zinfo->is_afs_inprogress = 1;
```

Comment:
instead of keeping this in zinfo, can this be moved to rebuild related structure?

Reply:
zvol_rebuild_scanner_info_t warg